### ISSUE TITLE
jQuery Migrate triggers deprecation warnings when using this plugin

### ISSUE THREAD

**Issue by stof** [harvesthq/chosen#2931]
When using this plugin alongside jQuery 3.2.1 and jQuery Migrate 3.0.1, we get a few deprecation warnings coming from Chosen:

- `jQuery.fn.focus() event shorthand is deprecated`
- `jQuery.fn.blur() event shorthand is deprecated`

The jQuery team advices to always use `.on()` and `.trigger()` instead in plugins, because shorthands can be excluded from jQuery when doing custom builds (and I think the official slim build excludes them).

### Environment

  - **Chosen Version**: 1.8.2
  - **jQuery Version**: 3.2.1, with jQuery Migrate 3.0.1 loaded (the development build, to get the warnings)

**Comment by tjschuck** [comment]
@stof We currently "support" as far back as jQuery 1.7, right?  Is there a fix here that will continue to work both for future versions of jQuery as well as back as far as that?  Or would we need to drop some old versions to move forward?

**Comment by stof** [comment]
Yes, using `.on('focus', ...)` instead (i.e. doing what the shortcut does internally). `.on` is available in 1.7+, so we would not need to change the supported jQuery versions.

Note that AFAIK, there is no plan for removing the shortcuts from the full build for now. However, there might be plans to promote the slim build more in the future.
We could add testing against the 3.x slim build on CI to ensure compat (in our case, I think we are affected only by the event shortcuts, as we don't use the AJAX and Effects APIs of jQuery)
